src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -423,16 +466,16 @@ grpc_status_code s2a_max_record_overhead(const s2a_crypter& crypter,
   GPR_ASSERT(max_record_overhead != nullptr);
   GPR_ASSERT(error_details != nullptr);
   size_t tag_size;
-  grpc_status_code status = s2a_tag_size(crypter, &tag_size, error_details);
+  grpc_status_code status =
+      s2a_tag_size(crypter.ciphersuite, &tag_size, error_details);
   if (status != GRPC_STATUS_OK) {
     return status;
   }
   *max_record_overhead = SSL3_RT_HEADER_LENGTH + tag_size + /* record type=*/1;
   return GRPC_STATUS_OK;
 }
 
-static grpc_status_code s2a_write_tls13_record_header(uint8_t record_type,
-                                                      size_t header_size,
+static grpc_status_code s2a_write_tls13_record_header(size_t header_size,
```

Comment:
If this API will only write application data record, maybe we should change the name?

Reply:
This function won't only be used to write application data records. All TLS 1.3 records, regardless of the record type, have the same record header. This common record header is the record header of a TLS 1.2 application data record.